src/components/Messages/PreviewList.tsx

```diff
@@ -30,51 +30,30 @@ interface Props {
 const PreviewList: FC<Props> = ({ className, selectedConversationKey }) => {
   const router = useRouter();
   const currentProfile = useAppStore((state) => state.currentProfile);
-  const messageProfiles = useMessageStore((state) => state.messageProfiles);
-  const setMessageProfiles = useMessageStore((state) => state.setMessageProfiles);
+  const addProfileAndSelectTab = useMessageStore((state) => state.addProfileAndSelectTab);
   const selectedTab = useMessageStore((state) => state.selectedTab);
   const setSelectedTab = useMessageStore((state) => state.setSelectedTab);
   const [showSearchModal, setShowSearchModal] = useState(false);
-  const { authenticating, loading, messages, profiles, profilesError } = useMessagePreviews();
+  const { authenticating, loading, messages, profilesToShow, requestedCount, profilesError } =
+    useMessagePreviews();
   const clearMessagesBadge = useMessagePersistStore((state) => state.clearMessagesBadge);
 
-  const sortedProfiles = Array.from(profiles).sort(([keyA], [keyB]) => {
+  const sortedProfiles = Array.from(profilesToShow).sort(([keyA], [keyB]) => {
     const messageA = messages.get(keyA);
     const messageB = messages.get(keyB);
     return (messageA?.sent?.getTime() || 0) >= (messageB?.sent?.getTime() || 0) ? -1 : 1;
   });
 
-  const followedProfiles = Array.from(sortedProfiles).filter(([, value]) => {
-    if (value.isFollowedByMe) {
-      return true;
-    }
-    return false;
-  });
-
-  const requestedProfiles = Array.from(sortedProfiles).filter(([, value]) => {
-    if (!value.isFollowedByMe) {
-      return true;
-    }
-    return false;
-  });
-
-  const profilesToShow = selectedTab === 'Following' ? followedProfiles : requestedProfiles;
-
   useEffect(() => {
     if (!currentProfile) {
       return;
     }
-    const profileKeys = Array.from(profiles.keys());
-    const messageKeys = Array.from(messages.keys());
-    const hasPreviews = profileKeys.some((item) => messageKeys.includes(item));
-    if (hasPreviews) {
-      clearMessagesBadge(currentProfile.id);
-    }
+    clearMessagesBadge(currentProfile.id);
```

Comment:
It should be ok to clear the badge here aggressively since we clear on click of the message icon as well.